inc/options/class-wpseo-options-backfill.php

```diff
@@ -0,0 +1,178 @@
+<?php
+/**
+ * @package WPSEO\Internals\Options
+ */
+
+/**
+ * Backfill the removed options in Yoast SEO 7.0.
+ *
+ * @since 7.0.2
+ */
+class WPSEO_Options_Backfill implements WPSEO_WordPress_Integration {
+	/** @var array List of options that have been removed. */
+	protected $lookup = array(
+		'wpseo_internallinks' => array(
+			'breadcrumbs-404crumb'      => 'breadcrumbs-404crumb',
+			'breadcrumbs-blog-remove'   => 'breadcrumbs-blog-remove',
+			'breadcrumbs-boldlast'      => 'breadcrumbs-boldlast',
+			'breadcrumbs-archiveprefix' => 'breadcrumbs-archiveprefix',
+			'breadcrumbs-enable'        => 'breadcrumbs-enable',
+			'breadcrumbs-home'          => 'breadcrumbs-home',
+			'breadcrumbs-prefix'        => 'breadcrumbs-prefix',
+			'breadcrumbs-searchprefix'  => 'breadcrumbs-searchprefix',
+			'breadcrumbs-sep'           => 'breadcrumbs-sep',
+		),
+		'wpseo_rss'           => array(
+			'rssbefore' => 'rssbefore',
+			'rssafter'  => 'rssafter',
+		),
+		'wpseo_xml'           => array(
+			'enablexmlsitemap'       => 'enable_xml_sitemap',
+			'disable_author_sitemap' => 'noindex-author-wpseo',
+			'disable_author_noposts' => 'noindex-author-noposts-wpseo',
+		),
+		'wpseo_permalinks'    => array(
+			'redirectattachment' => 'disable-attachment',
+			'stripcategorybase'  => 'stripcategorybase',
+		),
+	);
+
+	/**
+	 * Registers all hooks to WordPress.
+	 */
+	public function register_hooks() {
+		// Backfill options that were removed.
+		foreach ( $this->lookup as $option => $fields ) {
+			add_filter( 'pre_option_' . $option, array( $this, 'backfill_option' ), 10, 2 );
+		}
+
+		// Make sure renamed meta key is backfilled.
+		add_filter( 'get_user_metadata', array( $this, 'backfill_usermeta' ), 10, 3 );
+
+		// Extend the options that have removed items.
+		add_filter( 'option_wpseo_titles', array( $this, 'extend_wpseo_titles' ), 10, 1 );
+		add_filter( 'option_wpseo', array( $this, 'extend_wpseo' ), 10, 1 );
+	}
+
+	/**
+	 * Extends the WPSEO option with the removed option values.
+	 *
+	 * @param array $data The data of the option.
+	 *
+	 * @return array Modified data.
+	 */
+	public function extend_wpseo( $data ) {
+		// Make sure we don't get stuck in an infinite loop.
+		static $running = false;
+
+		// If we are already running, don't run again.
+		if ( $running ) {
+			return $data;
+		}
+		$running = true;
+
+		$keys = array(
```

Comment:
I would create a `private` getter function for this with a mention about which version we removed these keys in.